crates/iceberg-catalog/src/service/hooks.rs

```diff
@@ -0,0 +1,280 @@
+use std::{collections::HashMap, sync::Arc};
+
+use iceberg::TableIdent;
+use iceberg_ext::catalog::rest::{
+    CommitTransactionRequest, CommitViewRequest, CreateTableRequest, CreateViewRequest,
+    RegisterTableRequest, RenameTableRequest,
+};
+
+use crate::{
+    api::{
+        iceberg::{
+            types::{DropParams, Prefix},
+            v1::{DataAccess, NamespaceParameters, TableParameters, ViewParameters},
+        },
+        RequestMetadata,
+    },
+    service::{TableIdentUuid, ViewIdentUuid},
+    WarehouseIdent,
+};
+
+#[derive(Clone)]
+pub struct Hooks(Vec<Arc<dyn EndpointHooks>>);
+
+impl core::fmt::Debug for Hooks {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("Hooks").field(&self.0.len()).finish()
+    }
+}
+
+impl Hooks {
+    #[must_use]
+    pub fn new(hooks: Vec<Arc<dyn EndpointHooks>>) -> Self {
+        Self(hooks)
+    }
+}
+
+#[async_trait::async_trait]
+impl EndpointHooks for Hooks {
+    async fn commit_table(
+        &self,
+        _warehouse_id: WarehouseIdent,
```

Comment:
Why are we using underscores here when the parameter is used below?

Reply:
courtesy of intellij.. clippy complains about unused vars, refactor -> rename apparently ends up renaming it across trait impls as well